src/eckit/geo/Range.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+#include <vector>
+
+
+namespace eckit::geo {
+
+
+class Range {
+public:
+    // -- Constructors
+
+    Range(const Range&) = delete;
+    Range(Range&&)      = delete;
+
+    // -- Destructors
+
+    virtual ~Range() = default;
+
+    // -- Operators
+
+    Range& operator=(const Range&) = delete;
+    Range& operator=(Range&&)      = delete;
+
+    // -- Methods
+
+    size_t size() const { return n_; }
+    double a() const { return a_; }
+    double b() const { return b_; }
+    double eps() const { return eps_; }
+
+    virtual bool periodic() const { return false; }
+
+    virtual Range* flip() const                             = 0;
+    virtual Range* crop(double crop_a, double crop_b) const = 0;
```

Comment:
I'd add the `[[nodiscard]]` attribute here, or return `std::unique_ptr<Range>` better.